core/src/main/java/com/linecorp/armeria/common/metric/ExecutorServiceMetrics.java

```diff
@@ -0,0 +1,280 @@
+        if (executor instanceof ScheduledExecutorService) {
+            return monitor(registry, (ScheduledExecutorService) executor, executorServiceName, tags);
+        }
+        new ExecutorServiceMetrics(executor, executorServiceName, tags).bindTo(registry);
+        return new TimedExecutorService(registry, executor, executorServiceName, tags);
+    }
+
+    /**
+     * Record metrics on the use of an {@link ExecutorService}.
+     *
+     * @param registry            The registry to bind metrics to.
+     * @param executor            The executor to instrument.
+     * @param executorServiceName Will be used to tag metrics with "name".
+     * @param tags                Tags to apply to all recorded metrics.
+     * @return The instrumented executor, proxied.
+     */
+    public static ExecutorService monitor(MeterRegistry registry, ExecutorService executor,
+                                          String executorServiceName, Tag... tags) {
+        return monitor(registry, executor, executorServiceName, asList(tags));
+    }
+
+    /**
+     * Record metrics on the use of a {@link ScheduledExecutorService}.
+     *
+     * @param registry            The registry to bind metrics to.
+     * @param executor            The scheduled executor to instrument.
+     * @param executorServiceName Will be used to tag metrics with "name".
+     * @param tags                Tags to apply to all recorded metrics.
+     * @return The instrumented scheduled executor, proxied.
+     * @since 1.3.0
+     */
+    public static ScheduledExecutorService monitor(MeterRegistry registry, ScheduledExecutorService executor,
+                                                   String executorServiceName, Iterable<Tag> tags) {
+        new ExecutorServiceMetrics(executor, executorServiceName, tags).bindTo(registry);
+        return new TimedScheduledExecutorService(registry, executor, executorServiceName, tags);
+    }
+
+    /**
+     * Record metrics on the use of a {@link ScheduledExecutorService}.
+     *
+     * @param registry            The registry to bind metrics to.
+     * @param executor            The scheduled executor to instrument.
+     * @param executorServiceName Will be used to tag metrics with "name".
+     * @param tags                Tags to apply to all recorded metrics.
+     * @return The instrumented scheduled executor, proxied.
+     * @since 1.3.0
+     */
+    public static ScheduledExecutorService monitor(MeterRegistry registry, ScheduledExecutorService executor,
+                                                   String executorServiceName, Tag... tags) {
+        return monitor(registry, executor, executorServiceName, asList(tags));
+    }
+
+    private void monitor(MeterRegistry registry, @Nullable ThreadPoolExecutor tp) {
+        if (tp == null) {
+            return;
+        }
+
+        FunctionCounter.builder("armeria.executor.completed", tp, ThreadPoolExecutor::getCompletedTaskCount)
```

Comment:
If we are putting this class to `common`, it has to be generic, i.e. The caller should be able to specify the prefix, rather than hard-coded with `armeria.executor`.

Reply:
I thought we temporarily use it. But specifying the prefix sounds good. 👍